### ISSUE TITLE
[BUG]: Color contrast computation incorrect with shadow DOM

### ISSUE THREAD

**Issue by tombrunet** [IBMa/equal-access#687]
### Project

a11y checker extension

### Browser

Chrome

### Operating system

MacOS

### Description

Reported by Ariella Gilmore.

Checker is reporting contrast violations with the callouts. Looks like this may be computing something incorrectly due to shadow dom, but may be color tokens. Need more investigation.

### Steps to reproduce

* Open https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components/iframe.html?id=components-callout-quote--default

Two contrast violations are triggered. When using the color contrast analyzer, these both look fine. Something appears to not be calculating correctly.

**Comment by aliunwala** [comment]
From Triage:
![image](https://user-images.githubusercontent.com/4451128/155610393-b22811c4-d9d6-4e56-a0d4-2938aedcbd3c.png)
![image](https://user-images.githubusercontent.com/4451128/155610494-d863f3de-ef10-4919-955f-2ab0482196b2.png)

**Comment by tombrunet** [comment]
Recheck after #742

**Comment by marcjohlic** [comment]
Moving this back into Triage for validation now that #742 has been merged

**Comment by shunguoy** [comment]
the issue exists with two color contrast violations per version 3.1.25

**Comment by shunguoy** [comment]
probably walking a shadow dom for the wrong parents.

**Comment by shunguoy** [comment]
issue still the same with the fix in #826

**Comment by snidersd** [comment]
Investigating issue. Unable to create a test case. This only happens during runtime.

**Comment by shunguoy** [comment]
investigated the cause of the issue: the text elements are rendered under a shadow dom slot element, rather than in the light DOM.  The current color contrast logic doesn't support the shadow DOM tree.

**Comment by shunguoy** [comment]
creating local test cases to replicate and fix error

**Comment by shunguoy** [comment]
the error can be duplicated with a smaller shadow dom test without web component

**Comment by shunguoy** [comment]
The initial rule logic is written but needs to test in various cases.

**Comment by shunguoy** [comment]
test case: https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components/iframe.html?id=components-callout-quote--default.   A live test is required in a browser.

**Comment by eirckr** [comment]
@shunguoy violations are still present: 
<img width="1485" alt="Screen Shot 2022-07-29 at 1 33 12 PM" src="https://user-images.githubusercontent.com/62436670/181823055-5518806c-2de3-4775-97ca-c6e6df48c60f.png">

**Comment by shunguoy** [comment]
@eirckr  do you rebuild the rule server from the branch?

**Comment by eirckr** [comment]
Validated 
<img width="1479" alt="Screen Shot 2022-08-01 at 11 20 28 AM" src="https://user-images.githubusercontent.com/62436670/182196705-99d5125d-3cef-406e-bba2-bc0d1d7d3d19.png">
